R/utils.R

```diff
@@ -167,10 +167,20 @@ ask <- function(question, default = FALSE) {
 
 }
 
-proceed <- function(default = FALSE) {
-  ask("Do you want to proceed?", default = default)
+proceed <- function(prompt = interactive(), default = TRUE) {
+  if (prompt) {
+    ask("Do you want to proceed?", default = default)
+  } else {
+    default
+  }
 }
 
+check_can_proceed <- function(prompt = interactive()) {
```

Comment:
How do you feel about a different name; e.g.

```
halt(<value>)
cancel(<value>)
halt_if(<value>)
cancel_if(<value>)
```

Reply:
I'm happy to change to whatever you want; I just used `check_`, because that prefix is something we use reasonably commonly in r-lib/tidyverse to indicate that you can ignore the return value, because the main point of the function is to signal an error if needed. For example, `check_installed("pkg")` errors if pkg isn't installed, and `check_string(x)` errors if `x` isn't a string.